src/activation.jl

```diff
@@ -108,12 +108,18 @@ end
 
 See [Quadratic Polynomials Learn Better Image Features](http://www.iro.umontreal.ca/~lisa/publications2/index.php/attachments/single/205).
 """
-softsign(x) = x / (one(x) + abs(x))
+softsign(x::Real) = x / (one(x) + abs(x))
 
 
 """
     softplus(x) = log(exp(x) + 1)
 
 See [Deep Sparse Rectifier Neural Networks](http://proceedings.mlr.press/v15/glorot11a/glorot11a.pdf).
 """
-softplus(x) = ifelse(x > 0, x + log1p(exp(-x)), log1p(exp(x)))
+softplus(x::Real) = ifelse(x > 0, x + log1p(exp(-x)), log1p(exp(x)))
+
+# Provide an informative error message if activation functions are called with an array
+for f in (:σ, :σ_stable, :logσ, :relu, :leakyrelu, :elu, :gelu, :swish, :selu, :softsign, :softplus)
+  @eval $(f)(x::AbstractArray, args...) =
+    error("Use explicit invocations such as `", $(string(f)), ".(x)` to apply activation functions to tensors!")
```

Comment:
```suggestion
    error("Use broadcasting (`", $f, ".(x)`) to apply activation functions to arrays.")
```

I think `$f` will print the same but worth checking quickly.

Reply:
Actually in my initial implementation I used `$f`, but I added `string` since I was not satisfied with the output. With just `$f` I get, e.g., 
> ERROR: Use broadcasting (`NNlib.σ.(x)`) to apply activation functions to arrays.
whereas with `$(string(f))`
> ERROR: Use broadcasting (`σ.(x)`) to apply activation functions to arrays.
is printed.

Reply:
I see, a bit surprising that they behave differently, but that makes sense.